### PR TITLE
IN-2611: ensure content type of JSON before parsing sonar response

### PR DESCRIPTION
**Pull request by bsnyder788**

### CODE REVIEW

File: pkg/clients/tsclient/client.go [hunk 1]
```diff
@@ -35,6 +35,7 @@ import (
 
 const (
 	binaryContentType = "application/timeseries-binary-0"
+	jsonContentType   = "application/json"
```
Comment (thorfour) [inline pkg/clients/tsclient/client.go]: why are headers not already a const in the http package?

Comment (blockloop) [reply pkg/clients/tsclient/client.go]: I think there's a proposal but it was never merged

File: pkg/clients/tsclient/client.go [hunk 2]
```diff
@@ -443,7 +444,10 @@ func (c *HTTPClient) Flush() error {
 		}
 		return err
 	}
-	defer c.handleSonarResponse(resp.Body)
+	contentType := resp.Header.Get(contentTypeHeader)
+	if contentType == jsonContentType {
+		defer c.handleSonarResponse(resp.Body)
```
Comment (hoenn) [inline pkg/clients/tsclient/client.go]: Any idea why this was deferred in the first place?

Comment (bsnyder788) [reply pkg/clients/tsclient/client.go]: I'm not positive tbh

**Pull request merged**